subt/docker/robotika/ros/robotika/src/drone_height.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/python
+
+from __future__ import print_function
+import rospy
+from nav_msgs.msg import *
+from std_msgs.msg import *
+from sensor_msgs.msg import *
+from geometry_msgs.msg import *
+
+import pdb
+    
+HEIGHT = 1.5
+MAX_ANGULAR = 0.7
+lastScanDown = 0.0
+lastScanUp = 0.0
+lastOdom = None
+validOdom = True
+
+def scan_down_callback(scan):
+    global lastScanDown, publisherTwist
+    lastScanDown = scan
+
+def scan_up_callback(scan):
+    global lastScanUp, publisherTwist
+    lastScanUp = scan
+
+def twist_callback(cmd_vel):
+    global HEIGHT, lastScanDown, lastScanUp
+    
+    
+    if lastScanDown.ranges[0] > HEIGHT:
+        cmd_vel.linear.z = -0.1
+    else:
+        cmd_vel.linear.z = 0.1
```

Comment:
Doesn't this kind of control lead to oscillations? I would expect P control with maximum velocity clipping to work better. (And I would also not bother with PI or full PID.)